virttest/utils_test/__init__.py

```diff
@@ -1808,25 +1811,28 @@ def load_stress_tool(self):
         """
         load stress tool in guest
         """
-        session = self.get_session()
-        command = run_autotest(self.vm, session, self.control_path,
-                               None, None, self.params, copy_only=True,
-                               control_args=self.stress_args)
-        session.cmd("%s &" % command)
-        logging.info("Command: %s", command)
-        running = utils_misc.wait_for(self.app_running, first=0.5, timeout=60)
-        if not running:
-            raise StressError("Stress tool %s isn't running"
-                              % self.stress_type)
+        self.install()
+        session = self.session
+        session.cmd_output_safe('cd %s' % os.path.join(self.dst_path,
+                                                       self.base_name, self.work_path))
+        launch_cmds = 'nohup %s %s &' % (self.stress_cmds, self.stress_args)
+        error_context.context("Launch stress with command: %s" % launch_cmds)
+        try:
+            session.cmd_output_safe(launch_cmds)
+            # The background process sometimes does not return to
+            # terminate, if timeout, send a blank line afterward
+        except aexpect.ShellTimeoutError:
+            session.cmd_output_safe('')
+
+        return self.app_running()
```

Comment:
we need to assert whether stress is running or not, this would return Bool but caller doesn't know whether stress started properly or not.

Reply:
@balamuruhans Same, since this class has no `test` as its property, It only notice user whether the function be executed properly by return a bool value, the caller should has its own `test` object, which can raise the error there by itself. I assume that it is a 'function' class instead of a 'test' class.

Reply:
@vivianQizhu, IMHO your API `utils_test.VMStress` should be independent of the test or caller, we can use `exceptions.TestError()` because even if the test code or caller is not aware of, API should be consistent and do what it suppose to.

Reply:
@balamuruhans Yes, it makes sense. I will modify it. Thanks.